core/src/main/java/com/digitalpebble/stormcrawler/protocol/okhttp/HttpProtocol.java

```diff
@@ -267,6 +271,13 @@ private void addHeadersToRequest(Builder rb, Metadata md) {
                 rb.addHeader(h.getKey(), h.getValue());
             }
         }
+
+        final String agentName = md.getFirstValue(SET_USER_AGENT_BY_REQUEST, protocolMDprefix);
```

Comment:
what about having the logic in the _AbstractHTTPProtocol_ class and create a new method 
_String checkUAFromMD(Metadata md)_
?
If it returns non-null use it 

We can then have the same functionality in the other protocol implementations. I'd keep _"http.agent"_ as the key to look for in the metadata, put it in a static final String and use it in 

https://github.com/DigitalPebble/storm-crawler/blob/9b406a282cb4b1c8dff07074045e47918cecd280/core/src/main/java/com/digitalpebble/stormcrawler/protocol/AbstractHttpProtocol.java#L142